### ISSUE TITLE
Use XML documentation comments for GraphQL field comments

### ISSUE THREAD

**Issue by breyed** [EntityGraphQL/EntityGraphQL#230]
When generating schema documentation at runtime, such as in response to a web request, it is helpful to use custom attributes to embed type and field summaries into the DLL that contains the type structure. However, when only generating GraphQL files at design time, [XML documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/) are a better fit. They avoid adding unneeded documentation strings in production DLLs. They also provide IntelliSense to non-GraphQL code that uses the same symbols, such as descriptions for entity columns. 

To support XML documentation comments, `SchemaProvider<T>.ToGraphQLSchemaString` can take an optional parameter, which is an optional path to the XM file created by the C# compiler. For each GraphQL symbol, it would query the XML for the corresponding element and use its summary as the GraphQL comment.

**Comment by lukemurray** [comment]
Are you saying the the doc comments for the generated schema (via.`ToGraphQLSchemaString`) should use the C# doc comments for each of the mapped types etc?

Interesting idea, although we often have more internal comments in the code than we have in the "API" in the schema. Also I don't think that is available via reflection so I guess you're right it would need to be a path to the output file.

**Comment by breyed** [comment]
Yes, I'm suggesting to provide an option to use the C# doc comments for the comments for GraphQL generated schema. For projects where GraphQL is used internally, having an exact match between the C# comments and the comments in the generated .graphql file is desirable. Where the GraphQL files are exposed externally, a useful distinction can be to put public comments in the `<summary>` tag and detailed comments not exposed through GraphQL in the `<remarks>` or other tags.

You are correct that the C# comments are not available via reflection; they're in the DLL at all. They're just in the XML file generated by the C# compiler. They're designed for development-time use. Propagating the comments to the .graphql file extends their design-time presence from C# code into TypeScript (or the whatever language the front end is using). It helps the front end be a natural extension of the back end, which is largely the point of GraphQL in the first place.

**Comment by lukemurray** [comment]
closing this as not looking to implement. XML docs feel to internal vs. specific attributes etc. and looking at how we might make that easier. but this is complex with XML docs as builds can remove them, mostly Release builds. For now we'l stick to attributes and the fluent methods. and look at making them nicer.
